local_units/tasks.py

```diff
@@ -27,40 +28,40 @@ def send_local_unit_email(local_unit_id: int, users_details: list, new: bool = T
         email_subject = "Action Required: Local Unit Pending Validation"
         email_type = "Update Local Unit"
 
-    for email, first_name, last_name in users_details:
+    for user in users:
         # NOTE: Adding the validator email to the context
-        email_context["validator_email"] = email
-        email_context["full_name"] = f"{first_name} {last_name}"
+        email_context["validator_email"] = user.email
+        email_context["full_name"] = user.get_full_name()
         email_body = render_to_string("email/local_units/local_unit.html", email_context)
-        send_notification(email_subject, email, email_body, email_type)
+        send_notification(email_subject, user.email, email_body, email_type)
     return email_context
 
 
 @shared_task
-def send_validate_success_email(local_unit_id: int, user_id: int, new_or_updated: str = ""):
-    if not local_unit_id or not user_id:
+def send_validate_success_email(local_unit_id: int, message: str = ""):
+    if not local_unit_id:
         return None
 
-    user = User.objects.get(id=user_id)
     instance = LocalUnit.objects.get(id=local_unit_id)
+    user = User.objects.get(id=instance.created_by_id)
     email_context = get_email_context(instance)
     email_context["full_name"] = user.get_full_name()
     email_context["validate_success"] = True
     email_subject = "Your Local Unit Addition Request: Approved"
     email_body = render_to_string("email/local_units/local_unit.html", email_context)
-    email_type = f"{new_or_updated} Local Unit"
+    email_type = f"{message} Local Unit"
 
     send_notification(email_subject, user.email, email_body, email_type)
     return email_context
 
 
 @shared_task
-def send_revert_email(local_unit_id: int, user_id: int, reason: str = ""):
-    if not local_unit_id or not user_id:
+def send_revert_email(local_unit_id: int, reason: str = ""):
```

Comment:
Can we use the localunit to get the reason instead of passing it?

Reply:
We can but the revert reason is currently stored in different table.